core/jazz_es-kinesis-log-streamer/components/utils.js

```diff
@@ -136,93 +156,103 @@ function getApiLogsData(payload) {
-		}
-
-		domainAndservice = serviceInfo;
-		if(serviceInfo.indexOf(config.PATTERNS.sls_app_function) !== -1 ) { // if yes then sls-app function
-			let serviceInfoArr = serviceInfo.split(config.PATTERNS.sls_app_function);
-			domainAndservice = serviceInfoArr[0];
-		} 
-
-		logger.debug("domainAndservice: " + domainAndservice)
-    let domain = domainAndservice.substring(0, domainAndservice.indexOf("_"));
-    if (domain) {
-      data.domain = domain;
-      data.servicename = domainAndservice.substring(domain.length + 1);
-    } else {
-      data.domain = "";
-      data.servicename = domainAndservice;
-    }
-
-    if (data.servicename) {
-      payload.logEvents.forEach(function (logEvent) {
-
-        data.request_id = getSubInfo(logEvent.message, config.PATTERNS.guid_regex, 0);
-
-        data.platform_log_group = payload.logGroup;
-        data.platform_log_stream = payload.logStream;
-        data.timestamp = new Date(1 * logEvent.timestamp).toISOString();
-        let message = logEvent.message;
-        let messageLength = Buffer.byteLength(message, 'utf8');
-        if (messageLength > 32766) { //since 32766(32KB) is the default message size
-          let truncatedMessage = truncate(message, 32740); // message size + ...[TRUNCATED]
-          data.message = truncatedMessage + "  ...[TRUNCATED]";
+    data.asset_type = "lambda";
+    data.provider = "aws";
+    let domainAndservice, serviceInfo, environment, dev_environment, serviceInfoArr, domain;
+    data.request_id = getInfo(payload.logEvents, config.PATTERNS.lambda_request_id);
+    if (data.request_id) {
+      data.asset_identifier = payload.logGroup.split(config.PATTERNS.asset_identifier_key)[1];
+      // if tags present, then it is for sls-app, otherwise for other services
+      if(tagsResult != 'error' && tagsResult != undefined && tagsResult.tags.environment){
+        data.environment = tagsResult.tags.environment;
+        data.domain = tagsResult.tags.namespace;
+        data.servicename = tagsResult.tags.service;
+      } else {
+        environment = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment, 2);
+        if (environment === "dev") {
+          dev_environment = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment_dev, 2);
+          serviceInfo = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment_dev, 1);
+          data.environment = dev_environment;
         } else {
-          data.message = message;
+          data.environment = environment;
+          serviceInfo = getSubInfo(payload.logGroup, config.PATTERNS.lambda_domain_service, 1);
         }
 
-        data.log_level = getSubInfo(logEvent.message, config.PATTERNS.log_level, 0);
-        if (!data.log_level) {
-          data.log_level = config.DEFAULT_LOG_LEVEL;
+        domainAndservice = serviceInfo;
+        if(serviceInfo.indexOf(config.PATTERNS.sls_app_function) !== -1 ) { // if yes then sls-app function
```

Comment:
This block is not needed as we will not have the sls_app_function pattern in function name.

Reply:
Done.